### PR TITLE
docs: Improve  dynatrace config to drop empty attributes and resource attributes

### PR DESCRIPTION
**Pull request by rakesh-garimella**
## Description

Changes proposed in this pull request (what was done and why):

- There can be a case where attributes/resource attribute keys might have empty values. In such case these keys are dropped by the dynatrace backend (accepting the metric). There is warning in such a case which can cause confusion.

Unfortunately ottl does not support generic way of finding all the keys with values so currently hardcoding the keys which can have possible issue. 

Changes refer to particular issues, PRs or documents:

- https://github.com/kyma-project/telemetry-manager/issues/1317

## Traceability
- [ ] The PR is linked to a GitHub issue.
- [ ] The follow-up issues (if any) are linked in the `Related Issues` section.
- [ ] If the change is user-facing, the documentation has been adjusted.
- [ ] The feature is unit-tested.
- [ ] The feature is e2e-tested.

### CODE REVIEW

File: docs/user/integration/dynatrace/exporter-values.yaml [hunk 1]
```diff
@@ -39,8 +40,20 @@ config:
...
+          - delete_key(resource.attributes, "k8s.cluster.name") where resource.attributes["k8s.cluster.name"] == ""
+      - context: resource
+        statements:
+          # Remove empty attributes
+          - delete_key(attributes, "k8s.node.name") where attributes["k8s.node.name"] == ""
+          - delete_key(attributes, "k8s.namespace.name") where attributes["k8s.namespace.name"] == ""
+          - delete_key(attributes, "k8s.cluster.name") where attributes["k8s.cluster.name"] == ""
 
-    # Selects only one metric for the current collector pod 
+    # Selects only one metric for the current collector pod
```
Comment (NHingerl) [inline docs/user/integration/dynatrace/exporter-values.yaml:56]:
```suggestion
    # Selects only one metric for the current collector Pod
```

**Comment by NHingerl** [comment]
Why are tws codeowner for a yaml file tho?

File: docs/user/integration/dynatrace/exporter-values.yaml [hunk 2]
```diff
@@ -39,8 +40,20 @@ config:
...
+      - context: metric
+        statements:
+          # Remove empty resource attributes
+          - delete_key(resource.attributes, "k8s.node.name") where resource.attributes["k8s.node.name"] == ""
+          - delete_key(resource.attributes, "k8s.namespace.name") where resource.attributes["k8s.namespace.name"] == ""
+          - delete_key(resource.attributes, "k8s.cluster.name") where resource.attributes["k8s.cluster.name"] == ""
+      - context: resource
+        statements:
+          # Remove empty attributes
+          - delete_key(attributes, "k8s.node.name") where attributes["k8s.node.name"] == ""
```
Comment (a-thaler) [inline docs/user/integration/dynatrace/exporter-values.yaml:52]: Is that really meaningful? Will it ever happen that these attributes are not resource attributes.. and then are empty?

File: docs/user/integration/dynatrace/exporter-values.yaml [hunk 3]
```diff
@@ -39,8 +40,20 @@ config:
         statements:
           # convert the <histogram_name>_sum metrics to gauges.
           - convert_sum_to_gauge() where IsMatch(metric.name, ".*_sum")
+      - context: metric
+        statements:
+          # Remove empty resource attributes
+          - delete_key(resource.attributes, "k8s.node.name") where resource.attributes["k8s.node.name"] == ""
+          - delete_key(resource.attributes, "k8s.namespace.name") where resource.attributes["k8s.namespace.name"] == ""
+          - delete_key(resource.attributes, "k8s.cluster.name") where resource.attributes["k8s.cluster.name"] == ""
```
Comment (a-thaler) [inline docs/user/integration/dynatrace/exporter-values.yaml:48]: The cluster name cannot be empty. If it is the case, we must fix it..

File: docs/user/integration/dynatrace/exporter-values.yaml (hunk 2 above)
Comment (rakesh-garimella) [reply docs/user/integration/dynatrace/exporter-values.yaml]: removed

File: docs/user/integration/dynatrace/exporter-values.yaml (hunk 3 above)
Comment (rakesh-garimella) [reply docs/user/integration/dynatrace/exporter-values.yaml]: removed

**Pull request merged**